### PR TITLE
store reward information

### PR DESCRIPTION
**Pull request by bailantaotao**

### CODE REVIEW

File: migration/db/migrate/20180706033026_add_reward_information_to_header_tables.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class AddRewardInformationToHeaderTables < ActiveRecord::Migration[5.2]
+  def change
+    add_column :block_headers, :miner_reward, :string, :limit => 32, :null => false
+    add_column :block_headers, :uncles_inclusion_reward, :string, :limit => 32, :null => false
+    add_column :block_headers, :txs_fee, :string, :limit => 32, :null => false
+    add_column :block_headers, :uncles_reward, :string, :limit => 32, :null => true
+    add_column :block_headers, :uncle1_hash, :binary, :limit => 32, :null => true
+    add_column :block_headers, :uncle2_hash, :binary, :limit => 32, :null => true
+    add_index :block_headers, :uncle1_hash, :unique => true
```
[the remark this answers is not in the document]
Comment (bailantaotao) [reply migration/db/migrate/20180706033026_add_reward_information_to_header_tables.rb]: no, the `unique` has been removed!

File: common/utils.go [hunk 2]
```diff
@@ -195,3 +196,42 @@ func EthTransferEvent(b *types.Block, log *types.TransferLog) *model.Transfer {
...
+	big8  = big.NewInt(8)
+	big32 = big.NewInt(32)
+)
+
+// AccumulateRewards credits the coinbase of the given block with the mining
+// reward. The total reward consists of the static block reward and rewards for
+// included uncles. The coinbase of each uncle block is also rewarded.
+//
+// **COPIED FROM**: github.com/ethereum/go-ethereum/consensus/ethash/consensus.go#accumulateRewards()
+func AccumulateRewards(header *types.Header, uncles []*types.Header) (minerReward, minerBaseReward *big.Int, uncleReward []*big.Int, uncleHash []common.Hash) {
```
[the remark this answers is not in the document]
Comment (bailantaotao) [reply common/utils.go]: ok

File: model/types.go [hunk 3]
```diff
@@ -59,6 +68,24 @@ func (h Header) TableName() string {
...
+	totalUnclesReward := new(big.Int)
+	ush := make([][]byte, 2)
+	// Assume len(unclesReward) and len(unclesHash) are the same
+	for i, u := range unclesHash {
+		ush[i] = u.Bytes()
+		totalUnclesReward.Add(totalUnclesReward, unclesReward[i])
+	}
+
+	h.MinerReward = new(big.Int).Add(minerReward, txsFee).String()
+	h.UnclesInclusionReward = new(big.Int).Sub(minerReward, minerBaseReward).String()
```
[the remark this answers is not in the document]
Comment (bailantaotao) [reply model/types.go]: ok

File: common/utils.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kimmylin) [reply common/utils.go]: i was wrong. we don't have to care about coinbase in this PR. thanks

**Pull request merged**